### PR TITLE
Use the Python API server for some Java runtime calls

### PR DESCRIPTION
**Pull request by cdonati**
This removes the custom Java implementation for the following APIs:
- App Identity
 - Datastore
 - Memcache
 - Taskqueue

### CODE REVIEW

**Comment by cdonati** [comment]
The following PRs should be merged before this one:
- https://github.com/AppScale/appscale/pull/2986
- https://github.com/AppScale/appscale/pull/2987
- https://github.com/AppScale/appscale/pull/2989
- https://github.com/AppScale/appscale/pull/2993
- https://github.com/AppScale/appscale/pull/2994
- https://github.com/AppScale/appscale/pull/2995

File: AdminServer/appscale/admin/instance_manager/instance.py [hunk 1]
```diff
@@ -92,6 +92,20 @@ def create_java_start_cmd(app_name, port, load_balancer_host, max_heap,
...
   web_inf_directory = find_web_inf(revision_base)
 
+  api_server = os.path.join(SCRIPTS_DIR, 'appscale_java8_apiserver.sh')
+  apis_using_external_server = ['app_identity_service', 'datastore_v3',
+                                'memcache', 'taskqueue']
+  api_server_flags = [
+    '--application={}'.format(app_name),
+    '--datastore_path={}'.format(
+      ':'.join([options.db_proxy, str(DB_SERVER_PORT)])),
+    '--login_server={}'.format(load_balancer_host),
```
Comment (obino) [inline AdminServer/appscale/admin/instance_manager/instance.py]: Is load_balancer a single host? Does this point to login_host or to one of the internal load balancers?

File: AdminServer/appscale/admin/instance_manager/instance.py [hunk 2]
```diff
@@ -112,11 +126,17 @@ def create_java_start_cmd(app_name, port, load_balancer_host, max_heap,
...
-    "--api_using_python_stub=app_identity_service",
-    os.path.dirname(web_inf_directory)
+    "--path_to_python_api_server={}".format(api_server)
   ]
 
+  for flag in api_server_flags:
+    cmd.append('--python_api_server_flag="{}"'.format(flag))
+
+  for api in apis_using_external_server:
+    cmd.append('--api_using_python_stub={}'.format(api))
```
Comment (obino) [inline AdminServer/appscale/admin/instance_manager/instance.py]: Should it be "{}"?

File: scripts/appscale_java8_apiserver.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# Launches the AppScale Java 8 API server
+
+API_ARGS=""
+while test -n "${1}"; do
+  case "${1}" in
+    --application_prefix)
+      shift
+      ;;
+    --clear_datastore)
+      ;;
```
Comment (obino) [inline scripts/appscale_java8_apiserver.sh]: I assume the missing shift is on purpose: is this the only extra option we would like to pass on?

Comment (sjones4) [reply scripts/appscale_java8_apiserver.sh]: The matched arguments are dropped, when there is a related value it is also dropped. For `--clear_datastore` it is just a flag, so there is nothing else to drop.

File: AdminServer/appscale/admin/instance_manager/instance.py (hunk 1 above)
Comment (cdonati) [reply AdminServer/appscale/admin/instance_manager/instance.py]: `load_balancer_host` is the login property in this case.

File: AdminServer/appscale/admin/instance_manager/instance.py (hunk 2 above)
Comment (cdonati) [reply AdminServer/appscale/admin/instance_manager/instance.py]: None of the API service names have spaces, so I think it's safe to omit the quotes.

**Comment by cdonati** [comment]
http://ocd.appscale.net:8080/job/Daily%20Build/6208/
http://ocd.appscale.net:8080/job/Tools-test/4287/

**Pull request merged**